nano/node/nodeconfig.cpp

```diff
@@ -217,6 +223,26 @@ nano::error nano::node_config::deserialize_toml (nano::tomlconfig & toml)
 			});
 		}
 
+		if (toml.has_key ("new_work_peers"))
+		{
+			new_work_peers.clear ();
+			toml.array_entries_required<std::string> ("new_work_peers", [this](std::string const & entry) {
+				auto port_position (entry.rfind (':'));
+				bool result = port_position == -1;
+				if (!result)
+				{
+					auto port_str (entry.substr (port_position + 1));
+					uint16_t port;
+					result |= parse_port (port_str, port);
+					if (!result)
+					{
+						auto address (entry.substr (0, port_position));
+						this->new_work_peers.emplace_back (address, port);
+					}
+				}
```

Comment:
This is duplicated now, would be nice to get it extracted into a helper function, passing container and `entry`.